frontend/www/js/omegaup/components/course/Statistics.vue

```diff
@@ -0,0 +1,197 @@
+            option.text
+          }}</option>
+        </select>
+      </div>
+      <highcharts v-bind:options="selected"></highcharts>
+    </div>
+  </div>
+  <!-- panel -->
+</template>
+
+<script lang="ts">
+import { Chart } from 'highcharts-vue';
+import { Vue, Component, Prop, Watch } from 'vue-property-decorator';
+import { types } from '../../api_types';
+import T from '../../lang';
+import * as ui from '../../ui';
+@Component({
+  components: {
+    highcharts: Chart,
+  },
+})
+export default class Statistics extends Vue {
+  @Prop() course!: types.CourseDetails;
+  @Prop() problemStats!: types.CourseProblemStatistics[];
+  T = T;
+  //chart options
+  selected = this.varianceChartOptions;
+  options = [
+    { value: this.varianceChartOptions, text: T.courseStatisticsVariance },
+    { value: this.averageChartOptions, text: T.courseStatisticsAverageScore },
+    {
+      value: this.highScoreChartOptions,
+      text: T.courseStatisticsStudentsAbove,
+    },
+    {
+      value: this.lowScoreChartOptions,
+      text: T.courseStatisticsStudentsScored,
+    },
+    { value: this.minimumChartOptions, text: T.courseStatisticsMinimumScore },
+    { value: this.maximumChartOptions, text: T.courseStatisticsMaximumScore },
+  ];
+  //get chart options
+  get varianceChartOptions() {
+    return this.createChartOptions(
+      T.courseStatisticsScoreVariance,
+      '{y}',
+      T.courseStatisticsVariance,
+      this.getStatistic('variance'),
+      this.maxVariance,
+      this.problems,
+    );
+  }
+  get averageChartOptions() {
+    return this.createChartOptions(
+      T.courseStatisticsAverageScore,
+      '{y}',
+      T.wordsScore,
+      this.getStatistic('average'),
+      this.maxPoints,
+      this.problems,
+    );
+  }
+  get highScoreChartOptions() {
+    return this.createChartOptions(
+      T.courseStatisticsStudentsAbove,
+      '{y} %',
+      T.wordsPercentage,
+      this.getStatistic('high_score_percentage'),
+      100,
+      this.problems,
+    );
+  }
+  get lowScoreChartOptions() {
+    return this.createChartOptions(
+      T.courseStatisticsStudentsScored,
+      '{y} %',
+      T.wordsPercentage,
+      this.getStatistic('low_score_percentage'),
+      100,
+      this.problems,
+    );
+  }
+  get minimumChartOptions() {
+    return this.createChartOptions(
+      T.courseStatisticsMinimumScore,
+      '{y}',
+      T.wordsScore,
+      this.getStatistic('minimum'),
+      this.maxPoints,
+      this.problems,
+    );
+  }
+  get maximumChartOptions() {
+    return this.createChartOptions(
+      T.courseStatisticsMaximumScore,
+      '{y}',
+      T.wordsScore,
+      this.getStatistic('maximum'),
+      this.maxPoints,
+      this.problems,
+    );
+  }
+  //helper functions
+  get problems() {
+    let problems = [];
+    for (const problem in this.problemStats) {
+      problems.push(
+        `${this.problemStats[problem].assignment_alias} - ${this.problemStats[problem].problem_alias}`,
```

Comment:
the use of [`for...of`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of) is _almost_ always preferred over [`for...in`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in), since the latter has some [well-documented downsides](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in#Why_Use_for...in) (and produces smaller code to boot!):

```suggestion
    for (const problem of this.problemStats) {
      problems.push(
        `${problem.assignment_alias} - ${problem.problem_alias}`,
```

same in all the other `for` loops in this change (even the one in L134).

since the body of the loop is simple enough, it could also be rewritten as:

```typescript
    return this.problemStats.map((problem) =>
        `${problem.assignment_alias} - ${problem.problem_alias}`);
```